src/hooks/useSchemas.ts

```diff
@@ -0,0 +1,154 @@
+import { set } from "unchanged";
+import { useTrackedEffect } from "ahooks";
+
+import {
+  Order_By,
+  useDeleteSchemaMutation,
+  useExportDataMutation,
+  useCreateBranchMutation,
+  useCreateVersionMutation,
+  useSetDefaultBranchMutation,
+  useAllDataSchemasQuery,
+  useAllSchemasSubscription,
+} from "@/graphql/generated";
+import type {
+  Maybe,
+  AllSchemasSubscription,
+  AllDataSchemasQueryVariables,
+} from "@/graphql/generated";
+import CurrentUserStore from "@/stores/CurrentUserStore";
+
+import type { SubscriptionHandler } from "urql";
+
+type Pagination = Omit<AllDataSchemasQueryVariables, "where" | "order_by">;
+type Params = {
+  dataSourceId?: string;
+  teamId?: Maybe<string>;
+  statuses?: string[];
+};
+
+interface Props {
+  pauseQueryAll?: boolean;
+  pagination?: Pagination;
+  params?: Params;
+  disableSubscription?: boolean;
+}
+
+const getListVariables = (
+  pagination: Pagination,
+  params: Params = {}
+): AllDataSchemasQueryVariables => {
+  let res: AllDataSchemasQueryVariables = {
+    order_by: {
+      created_at: Order_By.Desc,
+    },
+  };
+
+  if (pagination) {
+    res = {
+      ...res,
+      ...pagination,
+    };
+  }
+
+  if (params?.dataSourceId) {
+    res = set("where.datasource_id._eq", params.dataSourceId, res);
+  }
+
+  if (params?.teamId) {
+    res = set("where.datasource.team_id._eq", params.teamId, res);
+  }
+
+  if (params?.statuses) {
+    res = set("where.status._in", params.statuses, res);
+  }
+
+  return res;
+};
+
+const handleSubscription: SubscriptionHandler<
+  AllSchemasSubscription,
+  AllSchemasSubscription
+> = (_, response) => response;
+
+export default (props: Props = {}) => {
```

Comment:
could we rename the file into `useModels`? We should not use `schemas` because it's confusing people, let's use `models`